Runtime/MeshChainRenderer.cs

```diff
@@ -0,0 +1,459 @@
+                colorKeys[lastColorIndex].color = flatColor;
+                alphaKeys[lastAlphaIndex].alpha = value.a;
+                m_Color.colorKeys = colorKeys;
+                m_Color.alphaKeys = alphaKeys;
+                UpdateColors();
+            }
+        }
+
+        /// <summary>
+        /// Resets the width to a straight curve at the given value
+        /// </summary>
+        /// <param name="newWidth">The new width for the curve to display</param>
+        public void SetTotalWidth(float newWidth)
+        {
+            m_Width = newWidth;
+            m_WidthCurve = new AnimationCurve(new Keyframe(0, 1.0f));
+            UpdateWidth();
+        }
+
+        /// <summary>
+        /// Resets the color to a single value
+        /// </summary>
+        /// <param name="newColor">The new color for the curve to display</param>
+        public void SetTotalColor(Color newColor)
+        {
+            var flatColor = newColor;
+            flatColor.a = 1.0f;
+            m_Color = new Gradient
+            {
+                alphaKeys = new[] { new GradientAlphaKey(newColor.a, 0), new GradientAlphaKey(newColor.a, 1), },
+                colorKeys = new[] { new GradientColorKey(flatColor, 0), new GradientColorKey(flatColor, 1) },
+                mode = GradientMode.Blend
+            };
+            UpdateColors();
+        }
+
+        void OnValidate()
+        {
+            SetupMeshBackend();
+            if (NeedsReinitialize())
+            {
+#if UNITY_EDITOR
+                UnityEditor.EditorApplication.delayCall += EditorCheckForUpdate;
+#endif
+            }
+            else
+            {
+                EditorCheckForUpdate();
+            }
+        }
+
+        /// <summary>
+        /// Cleans up the visible interface of the MeshRenderer by hiding unneeded components
+        /// Also makes sure the animation curves are set up properly to defaults
+        /// </summary>
+        void SetupMeshBackend()
+        {
+            if (m_MeshRenderer == null)
+                m_MeshRenderer = GetComponent<MeshRenderer>();
+
+            if (m_MeshRenderer.hideFlags == HideFlags.None)
+                m_MeshRenderer.hideFlags = HideFlags.HideInInspector;
+
+            var meshFilter = GetComponent<MeshFilter>();
+            if (meshFilter.hideFlags == HideFlags.None)
+                meshFilter.hideFlags = HideFlags.HideInInspector;
+
+            if (m_Materials == null || m_Materials.Length == 0)
+            {
+                m_Materials = m_MeshRenderer.sharedMaterials;
+            }
+
+            var sharedMaterials = m_MeshRenderer.sharedMaterials;
+            var length = sharedMaterials.Length;
+            var materialsEqual = true;
+            for (var i = 0; i < length; i++)
+            {
+                if (sharedMaterials[i] != m_Materials[i])
+                {
+                    materialsEqual = false;
+                    break;
+                }
+            }
```

Comment:
@AndrewTHEManeri this is unfortunately complex, but I needed to do this to prevent setting `.sharedMaterials` on every OnValidate. Is there some other way to make sure that `m_Materials` matches `sharedMaterials`? Is it enough to just check if `m_Materials` is null/empty?

Reply:
That should be fine.  Honestly in 99% of the cases it will be a one element loop